taichi/rhi/vulkan/vulkan_device_creator.h

```diff
@@ -4,7 +4,11 @@
 #define VK_USE_PLATFORM_WIN32_KHR 1
 #endif
 
-#include "taichi/rhi/vulkan/vulkan_common.h"
+#ifndef VK_NO_PROTOTYPES
```

Comment:
Better put the macro definitions with global effects in a common place like `rhi_macros.h` and make sure that file is included whatsoever. Otherwise, compile units with `vulkan_device_creator.h` included may have different VK_NO_PROTOTYPES definition compared to the other files.